src/services/createRouter.ts

```diff
@@ -171,7 +172,7 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
   function install(app: App): void {
     app.component('RouterView', RouterView)
     app.component('RouterLink', RouterLink)
-    app.provide(routerInjectionKey, router)
+    app.provide(routerInjectionKey, router as RegisteredRouter)
```

Comment:
this cast makes sense. `router` is `Router<T>` and the injection key expected registered router